plot/plot3d/clip: Simplify the -A syntax to -Ax|y

seisman opened the pull request:
See https://github.com/GenericMappingTools/gmt/issues/8645 for context.

Closes https://github.com/GenericMappingTools/gmt/issues/8645.

remkos reviewed (approved):
I felt I should review this since I implemented -Am and -Ap in GMT4 in Sep 2007!
Since then it was extended with -A[x|y|r|t], without it actually requiring a change of code except for needing to parse the options and add to the documentation.

So it is OK to me to bring it back to -A[x|y] as long as the old options are still supported for backward compatibility.

How does this impact the long options?

seisman commented:
> I felt I should review this since I implemented -Am and -Ap in GMT4 in Sep 2007!

Good to know the history dating back to 2007!

When updating the documentation, we also found a bug that the meaning of `-Am` and `-Ap` are reversed. Please see https://github.com/GenericMappingTools/gmt/issues/8645#issuecomment-2566198810 for details. We already fixed it in #8648, but it would be great if you can verify it again.

> So it is OK to me to bring it back to -A[x|y] as long as the old options are still supported for backward compatibility.

Yes, it's backward compatible.

> How does this impact the long options?

That's a good question. Currently, long names exist for m/p/x/y/r/t (https://github.com/GenericMappingTools/gmt/blob/605da462cbc15e3db40cc86e1e3b863b61444ca8/src/longopt/psxy_inc.h#L29). I guess we should also remove the long names for m/p/r/t, OK?

joa-quim commented:
Also -E longoptions got it all wrong GMT.jl has `error_bars`, PyGMT `error_bar` but GMT picked `errorbars`. Add all three aliases in master.

seisman commented:
Not yet. There are other modules (e.g., `clip`) that takes the same `-A` option.

seisman commented:
I believe this PR is now ready for final review.

joa-quim reviewed (approved):
Looks good.
